providers/namecheap/namecheap.go

```diff
@@ -18,9 +20,136 @@ type Namecheap struct {
 
 func init() {
 	providers.RegisterRegistrarType("NAMECHEAP", newReg)
+	providers.RegisterDomainServiceProviderType("NAMECHEAP", newDsp)
 }
 
-func newReg(m map[string]string) (providers.Registrar, error) {
+func newDsp(conf map[string]string, metadata json.RawMessage) (providers.DNSServiceProvider, error) {
+	return newProvider(conf, metadata)
+}
+
+func newReg(conf map[string]string) (providers.Registrar, error) {
+	return newProvider(conf, nil)
+}
+
+func splitDomain(domain string) (string, string) {
+	ds := strings.Split(domain, ".")
+	dl := len(ds)
+	tld := ds[dl-1]
```

Comment:
This method fails for tlds like `.co.uk'. Should use https://godoc.org/golang.org/x/net/publicsuffix instead.

Its kinda crappy that their api expects SLD/TLD, and https://www.namecheap.com/support/api/methods/domains/get-info.aspx doesn't even split it like that for you.